### PR TITLE
Bug/get bookings by period of time

### LINKED ISSUE #28: Missing: get bookings by period of time
Add a parameter to the GET /api/v1/Booking request to obtain only the reservations that exist in the indicated time period.

### PR DESCRIPTION
**Pull request by KsquarianMike**
Closes #28

### CODE REVIEW

File: app/controllers/v1/Booking.ts [hunk 1]
```diff
@@ -511,47 +536,60 @@ export class BookingController extends Controller {
...
-
-        if (bookings) {
-          const parsedBookings = JSON.parse(JSON.stringify(bookings));
-          const finalBookings = await this.bookingsPlusAttendees(
-            parsedBookings
-          );
-          const DTOBookings = finalBookings.map(bookingMapper.toDTO);
-          return res.status(200).json(DTOBookings);
-        }
+      const isEmptyFromDate = isEmpty(data.query.fromDate);
```
Comment (mishelashala) [inline app/controllers/v1/Booking.ts]: pass just the `predicate`, there's no need to have an assigment

File: app/controllers/v1/Booking.ts [hunk 2]
```diff
@@ -511,47 +536,60 @@ export class BookingController extends Controller {
...
-          const DTOBookings = finalBookings.map(bookingMapper.toDTO);
-          return res.status(200).json(DTOBookings);
-        }
       }
 
-      return Controller.badRequest(
-        res,
-        "Bad Request: fromDate must be a date in format YYYY-MM-DDTHH:MM."
-      );
+      const parsedBookings = JSON.parse(JSON.stringify(bookings));
```
Comment (mishelashala) [inline app/controllers/v1/Booking.ts]: why do you need this?

File: app/controllers/v1/Booking.ts [hunk 3]
```diff
@@ -511,47 +536,60 @@ export class BookingController extends Controller {
...
+          "Bad Request: toDate must be a date in format YYYY-MM-DDTHH:MM."
+        );
       }
 
-      // Obtain all bookings from a date
-      else if (isValidDate(toDate)) {
-        const bookings = await this.model.findAll({
+      let bookings;
+
+      if (isEmptyFromDate && isEmptyToDate) {
```
Comment (mishelashala) [inline app/controllers/v1/Booking.ts]: Create an abstraction for this, you can add a `strategy` function.

```js
const dateRangeStrategy = ({ fromDate, toDate }) => {
  if (...) {
    return ...
  } else if (...) {
    return ...
  } else {
    return ...
  }
}

const bookings = dateRangeStrategy({ fromDate, toDate });
```

File: app/controllers/v1/Booking.ts (hunk 2 above)
Comment (KsquarianMike) [reply app/controllers/v1/Booking.ts]: parsedBookings? it's useful for delete all the metadata that contain "bookings" ("bookings" is the result of the query to MySQL)

Comment (mishelashala) [reply app/controllers/v1/Booking.ts]: Why do you need to stringify and the to parse?

Comment (KsquarianMike) [reply app/controllers/v1/Booking.ts]: it's the same that doing booking[index].toJSON() for each booking in bookings, the method toJSON must have what data will be serialized, with this all the metada are removed.

**Pull request merged**